Fix #2057: improve the display of collection node list editor when there are too many nodes.

amgowano opened the pull request:
@BenHenning: PTAL.

seanlip commented:
Thanks, @amgowano! I took a brief look at the UI, and have some thoughts:

- I think you should keep the scrollbar. Otherwise, there's no visual affordance for how to scroll, and it took me a bit of guessing to figure out that I had to ... essentially pan, although the standard arrow cursor didn't signal this.
- The outer card should have a max width -- past a certain point, it gets uncomfortable to read. Note that we do something similar already for other places like the profile page -- past a certain point, we stop expanding the card, because reading something that wide would be uncomfortable. I think it makes sense to do that here, too.

seanlip commented:
/cc @kevinlee12

BenHenning commented on core/templates/dev/head/collection_editor/editor_tab/collection_main_tab_directive.html:
```diff
@@ -52,6 +64,10 @@
     width: 98%;
   }
 
+  .collection-editor-node-list-container::-webkit-scrollbar {
```
Isn't this specific to Chrome/Safari? What about IE/Firefox/Opera?

amgowano replied:
Removed.

amgowano replied:
Removed this and keep the scrollbar instead.

BenHenning commented:
Thanks @amgowano. The code looks fine. Regarding the UI, I was rather confused how to scroll, too. Will this approach work on mobile? I'm also not keen on using only keyboard controls for scrolling, especially since we don't have strong support for keyboard controls in either editor. It's fine to use arrow keys (really awesome, actually), but I think we need to have a fall-back for mouse-only control.

BenHenning commented:
Hi @amgowano did you get a chance to address our review comments? Thanks!

seanlip commented:
Hi @amgowano -- any plans to finish this? (Just doing the usual sweep of PRs in preparation for this month's release.)

amgowano commented:
@seanlip: Sorry I didn't have time to work on this to meet last month's release. I was finishing another issue. I am working on it now.

BenHenning commented:
Hi @amgowano, how's this going?

amgowano commented:
Sorry this unnecessarily took forever. I have addressed the comments as follows:

1. I have kept the scrollbar as per Sean's comments.
2. I have limited the width of the outer card to follow the limit that is imposed by Oppia library carousel. (However I feel like this is a disadvantage since this makes use introduce scrolling too early).

Also Ben the scrolling for mobile I believe should work by swiping.

BenHenning commented:
Replaced by #2422.

seanlip commented:
Closing this in favour of the new PR, #2422.
